scripts/test_grok1_block_forward.py

```
@@ -327,6 +329,44 @@ def test_matches_naive_per_head_reference(self):
+                got = self._call(chunk_size=chunk_size)
+                self.assertEqual(got.shape, (self.tokens, self.width))
+                np.testing.assert_allclose(got, expect, rtol=1e-4, atol=1e-4)
+
+    def test_chunk_size_is_clamped_to_the_documented_ceiling(self):
+        """A caller cannot opt out of the bound the chunking exists to enforce."""
+        from grok1_block_forward import ATTENTION_CHUNK_TOKENS
+
+        seen: list[int] = []
+        real_softmax = grok1_block_forward._softmax_fp32
+
+        def spy(logits):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `spy`

(ANN202)